### PR TITLE
Feat: set the initial content of config via environment variables & corrected `RedisConfig.DB` to int type

### PR DESCRIPTION
**Pull request by AH-dark**
In containerized deployments and Kubernetes cluster deployments, using `conf.ini` configuration is unnecessary, and in most cases we use environment variables instead of configuration files. 
Therefore, adaptation to environment variables config is necessary.

## Env Configuration

In commit cc303e74ddec232c7743406f36eda1a2f914cdda, I created `env.go` in util package, which provides an better method for obtaining environment variables. This way, we can fully resolve the environment variables in the init section, while having no conflict on existing configuration.

```go
// EnvStr returns the value of the environment variable named by the key.
func EnvStr(key, defaultValue string) string {
    if value, exist := os.LookupEnv(key); exist {
        return value
    }

    return defaultValue
}
```

## DB property from String to Int

As we all know, convert string value to int is more danger than convert int value to string, so I changed the type of `RedisConfig.DB` and some configuration in commit 50c3100afb79284a2c596c60cae291cc8aa2f671. 

Existing configurations will not be affected, and instead of handling errors for the `strconv.Atoi()` function, we use `strconv.Itoa()` without any error throwing.

```go
// Old
db, err := strconv.Atoi(database)
if err != nil {
    return nil, err
}

c, err := redis.Dial(
    network,
    address,
    redis.DialDatabase(db),
    redis.DialPassword(password),
)

// New
c, err := redis.Dial(
    network,
    address,
    redis.DialDatabase(database),
    redis.DialPassword(password),
)
```

## Skip configuration loading

In a container, we may not be able to get read and write permissions for the current directory. 
Therefore, we don't need to still generating and reading the `conf.ini` file in the mode of getting the configuration via environment variables.

So, I commited some code in commit cc303e74ddec232c7743406f36eda1a2f914cdda, which allowed using flag `--skip-conf` to skip configuration loading and generating.

### CODE REVIEW

**Comment by AH-dark** [comment]
The test ci failed due to gomod conflicts, so should I make another commit for `go mod tidy` ?

File: pkg/conf/defaults.go [hunk 1]
```diff
@@ -1,52 +1,60 @@
...
-	Charset: "utf8",
-	DBFile:  "cloudreve.db",
-	Port:    3306,
+	Type:        util.EnvStr("DATABASE_TYPE", "UNSET"),
+	User:        util.EnvStr("DATABASE_USER", "root"),
+	Password:    util.EnvStr("DATABASE_PASSWORD", ""),
+	Host:        util.EnvStr("DATABASE_HOST", "localhost"),
+	Name:        util.EnvStr("DATABASE_NAME", "cloudreve"),
+	TablePrefix: util.EnvStr("DATABASE_TABLE_PREFIX", ""),
+	DBFile:      util.EnvStr("DATABASE_DBFILE", "data.db"),
```
Comment (HFO4) [inline pkg/conf/defaults.go:21]:
```suggestion
	DBFile:      util.EnvStr("DATABASE_DBFILE", "cloudreve.db"),
```

File: pkg/util/env.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+import (
+	"os"
+	"strconv"
+	"strings"
+)
+
+// EnvStr returns the value of the environment variable named by the key.
+func EnvStr(key, defaultValue string) string {
+	if value, exist := os.LookupEnv(key); exist {
```
Comment (HFO4) [inline pkg/util/env.go:11]: I suggest adding a `CD_` prefix.

Comment (AH-dark) [reply pkg/util/env.go]: I completed it on https://github.com/cloudreve/Cloudreve/pull/1475/commits/bba4e7044a1491e7162e03dd0c356eb599d322e8

**Comment by AH-dark** [comment]
There are still problems, `HashIDSalt` and `SessionSecret` may be empty, which may bring security issues, but I don't have a better solution.

**Review by topjohncian — CHANGES_REQUESTED** [review]
Great job!

Perhaps we missed one thing - initializing the default config with generating `HashIDSalt` and `SessionSecret` at the first starting.

My opinion is to generate a `conf.ini` with `HashIDSalt` and `SessionSecret` even though skips conf.

But it creates `conf.ini`. 😿

**Comment by AH-dark** [comment]
> Great job!
> 
> Perhaps we missed one thing - initializing the default config with generating `HashIDSalt` and `SessionSecret` at the first starting.
> 
> My opinion is to generate a `conf.ini` with `HashIDSalt` and `SessionSecret` even though skips conf.
> 
> But it creates `conf.ini`. 😿

I just became aware of this issue and made a comment at https://github.com/cloudreve/Cloudreve/pull/1475#issuecomment-1262415549 but I am missing a solution.

**Comment by AH-dark** [comment]
> Great job!
> 
> Perhaps we missed one thing - initializing the default config with generating `HashIDSalt` and `SessionSecret` at the first starting.
> 
> My opinion is to generate a `conf.ini` with `HashIDSalt` and `SessionSecret` even though skips conf.
> 
> But it creates `conf.ini`. 😿

I tried to use random string in default value and it passed my simple test, but I don't know if there is an underlying problem.

**Comment by HFO4** [comment]
Clean up stale PR, feel free to reopen it if needed.

**Pull request closed without merging**